### PR TITLE
Gains dataset `abcd_demo`, begin deprecation of `ald_demo`

### LINKED ISSUE #279: Deprecate references to ALD in favour of ABCD
See this brief blog post explaining the decision: https://2degreesinvesting.github.io/posts/2022-03-02-ald-becomes-abcd/

In `r2dii.data` the key placement of this change will be in the dataset `ald_demo` which will become `abcd_demo`, as well as a column within that dataset `ald_demo$ald_timestamp` -> `abcd_demo$abcd_timestamp`. 

As a first step, we should add this new dataset, `abcd_demo` (which just contains identical data to `ald_demo`), and add a deprecation warning to `ald_demo`.

### PR DESCRIPTION
**Pull request by jdhoffa**
In this PR I begin the depracation process of the dataset `ald_demo`, in favour of the dataset `abcd_demo`. This has been done by: 
- Adding a new dataset, `abcd_demo`, that contains identical data to `ald_demo`
- Adding a `superseded` badge to the documentation of `ald_demo`

I wanted to also add a warning to `ald_demo` (using `lifecycle::deprecate_soft()`), however I wasn't sure how to do this, since we are working with a dataset, and not a function (there doesn't seem to be anywhere to embed the call to `lifecycle::deprecate_soft()`). I also couldn't find any guidance on the lifecycle website, @maurolepore any comments? 

Closes #279

### CODE REVIEW

**Comment by maurolepore** [comment]
@jdhoffa I added a warning via `delayAssign()`. Note I removed `ald_demo` from data/ because we no longer need it: We assign `ald_demo <- abcd_demo` on the fly via `delayAssign()`.

Feel free to remove my commits. I now realized I should have created a PR off your PR. Let me know if you prefer me to revert and do that instead.

``` r
library(r2dii.data)

# Note the warning
identical(ald_demo, abcd_demo)
#> Warning: `ald_demo` was superseded in r2dii.data 0.3.0. Please use `abcd_demo`
#> instead.
#> [1] TRUE

# After the first warning in a session, it doesn't warn 
# I don't know why but it's kind of nice
identical(ald_demo, abcd_demo)
#> [1] TRUE
```

<sup>Created on 2022-04-26 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by jdhoffa** [comment]
> @jdhoffa I added a warning via `delayAssign()`. Note I removed `ald_demo` from data/ because we no longer need it: We assign `ald_demo <- abcd_demo` on the fly via `delayAssign()`.
> 
> Note the warning assumes the next release will have version 0.3.0. Feel free to adapt.
> 
> Feel free to remove my commits. I now realized I should have created a PR off your PR. Let me know if you prefer me to revert and do that instead.
> 
> ```r
> library(r2dii.data)
> 
> # Note the warning
> identical(ald_demo, abcd_demo)
> #> Warning: `ald_demo` was superseded in r2dii.data 0.3.0. Please use `abcd_demo`
> #> instead.
> #> [1] TRUE
> 
> # After the first warning in a session, it doesn't warn 
> # I don't know why but it's kind of nice
> identical(ald_demo, abcd_demo)
> #> [1] TRUE
> ```
> 
> Created on 2022-04-26 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)

Thanks @maurolepore this works awesome. And that's ok that you pushed directly to this PR. 
I do plan on this change being bundled in `r2dii.data 0.3.0` so that works well. 

I think this PR is ready for a proper review now then.

File: DESCRIPTION [hunk 1]
```diff
@@ -38,6 +28,7 @@ BugReports: https://github.com/2DegreesInvesting/r2dii.data/issues
 Depends: 
     R (>= 3.4)
 Imports:
+    lifecycle,
```
Comment (maurolepore) [inline DESCRIPTION:31]: Is it still worth depending on lifecycle? The warning we implemented in `delayAssign()` does not use it.

File: R/ald_demo.R [hunk 2]
```diff
@@ -1,16 +1,19 @@
 #' An asset level dataset for demonstration
 #'
 #' @description
+#' `r lifecycle::badge("superseded")`
```
Comment (maurolepore) [inline R/ald_demo.R:4]: Yeah, this IS nice but if it's the only place we use lifecycle maybe it's not worth gaining a dependency.

File: R/ald_demo.R [hunk 3]
```diff
@@ -1,16 +1,19 @@
...
+#'
 #' Fake data about physical assets (e.g. wind turbine power plant
 #' capacities) used to assess the climate alignment of financial portfolios.
 #' It imitates data from market-intelligence databases.
 #'
 #' @template info_demo-datasets
 #'
+#' @keywords internal
```
Comment (maurolepore) [inline R/ald_demo.R:12]: Maybe we can delete R/ald_demo.R and use `#' @aliases` in `abcd_demo`?

File: data-raw/abcd_demo.R [hunk 4]
```diff
@@ -168,7 +168,9 @@ ordered_names <- c(
   "ald_timestamp",
   emission_factor_unit = "ald_emission_factor_unit"
 )
+# FIXME: Remove all references to abcd_demo from this file
```
Comment (maurolepore) [inline data-raw/abcd_demo.R:171]: SORRY! I meant to say 

FIXME: Remove all references to `ald_demo` from this file. 

We no longer need `ald_demo`  -- it's assigned on the fly if and when the user calls `ald_demo`.

File: data/ald_demo.R [hunk 5]
```diff
@@ -0,0 +1,14 @@
+delayedAssign("ald_demo", {
+  on_r_cmd <- !identical(Sys.getenv("R_CMD"), "")
+  if (!on_r_cmd) {
+    warning(
+      "`ald_demo` was superseded in r2dii.data 0.3.0. ",
+      "Please use `abcd_demo` instead.",
+      "\n Read this blog post to learn more:",
+      "\n https://2degreesinvesting.github.io/posts/2022-03-02-ald-becomes-abcd/",
```
Comment (maurolepore) [inline data/ald_demo.R:8]: Running this in my head I think these two lines will be indented by one character. Is this what you want?

**Review by maurolepore — APPROVED** [review]
LGTM but see my comments. In general I would do one more pass looking for things to prune -- so we don't leave things behind. For example, we no longer need to export `ald_demo` so there is no point in creating it in data-raw/ or having a .csv file.

**Comment by jdhoffa** [comment]
> LGTM but see my comments. In general I would do one more pass looking for things to prune -- so we don't leave things behind. For example, we no longer need to export `ald_demo` so there is no point in creating it in data-raw/ or having a .csv file.

Ok fair enough, I will. If I'm honestly, I've been a little conflicted about how to handle this update. 
Part of me wants to keep the PRs very focused and small (ie. ONLY add `abcd_demo`, and warn that `ald_demo` will be removed, and not change much else). The other side of me wants to be complete, and remove EVERYTHING related to `ald_demo` except for whatever is necessary to keep it working until it is fully deprecated. 

It sounds like you prefer the latter approach.

File: DESCRIPTION (hunk 1 above)
Comment (jdhoffa) [reply DESCRIPTION:31]: I think we still need to depend on it for the badges it provides??

File: R/ald_demo.R (hunk 2 above)
Comment (jdhoffa) [reply R/ald_demo.R:4]: So, knowing that we will be introducing several other `lifecycle` dependencies in `r2dii.match` and `r2dii.analysis` (mainly for the badges on arguments that will be lost), I am not terribly opposed to adding it here as well. 

`r2dii.data` is basically useless without those packages as well, so they will need to install that dependency at some point.

File: R/ald_demo.R (hunk 3 above)
Comment (jdhoffa) [reply R/ald_demo.R:12]: Hmm, ok so two options: 
* Keep `R/ald_demo.R` and `R/abcd_demo.R`. Benefit here is that, we can keep the documentation of `ald_demo` internal (so it is hard to find), but if someone does try to pull up the documentation directly, they will see immediately that the dataset is superseded (by the lifecycle badge) 
* Make `ald_demo` an alias of `R/abcd_demo.R`. Benefit here is that there is less code relating to `ald_demo`. Downside is that if people run `?ald_demo`, they wont be notified that it is superseded, and they may not even notice that they are looking at the documentation to `abcd_demo`.

Comment (jdhoffa) [reply R/ald_demo.R:12]: Personally I tend to prefer that if people lookup `?ald_demo` they get a big fat warning that it is superseded... but I'm not opposed to the `@alias` approach if you think it is cleaner.

File: data/ald_demo.R (hunk 5 above)
Comment (jdhoffa) [reply data/ald_demo.R:8]: The `\n ` character (with the space) does not indent the text anymore than the rest of the warning: 
Screenshot 2022-04-28 at 14 45 11

**Comment by jdhoffa** [comment]
> For example, we no longer need to export `ald_demo` so there is no point in creating it in data-raw/ or having a .csv file.

Also it isn't created in `data-raw/` and there is no `.csv` anymore? These had already been removed. The only place it appears still is in the data-dictionary definition, which i will remove it from.

File: R/ald_demo.R (hunk 2 above)
Comment (maurolepore) [reply R/ald_demo.R:4]: Good point.

**Comment by maurolepore** [comment]
Yeah, I think your idea of keeping the PR small is totally valid. Two advantages of the alternative that I see:

1. Get closer to meeting the legal obligation of removing references to "ald"
2. Get stuff done while you have the context in your head. No big deal -- as you can search easily for file-names and text containing "ald".

File: DESCRIPTION (hunk 1 above)
Comment (maurolepore) [reply DESCRIPTION:31]: Yeah, the badges are nice. Maybe once you complete the deprecation you may even remove the dependency on lifecycle. (So far this package imports nothing that does not come with R. There are a few suggested packages but nothing in imports other than utils and stats.)

File: R/ald_demo.R (hunk 3 above)
Comment (jdhoffa) [reply R/ald_demo.R:8]: Ok, I've minimized the documentation, but still using the lifecycle bade. Let me know what you think.

File: DESCRIPTION (hunk 1 above)
Comment (jdhoffa) [reply DESCRIPTION:31]: Ya, that's a great idea.

**Review by AlexAxthelm — APPROVED** [review]

**Pull request merged**